src/app/login/login.component.ts

```diff
@@ -95,4 +95,10 @@ export class LoginComponent implements OnInit {
         this.reRoute();
       }, (error) => this.message = 'Username and/or password do not match');
   }
+  loginAndEditProfile({ name, password }: {name: string, password: string}) {

```

Comment:
Instead of creating a new function which is almost identical to the `login` function, let's refactor the `login` function to take an optional parameter which when set will redirect to the user's update page.